src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -128,35 +128,46 @@ default void waitOnBuild(IClient client, long startTime, String bldId, TaskListe
                 listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);
 
             if (needToFollow.compareAndSet(true, false)) {
-                final String container = pod.getContainers().iterator().next().getName();
-
-                stop = pod.accept(new CapabilityVisitor<IPodLogRetrievalAsync, IStoppable>() {
-                    @Override
-                    public IStoppable visit(IPodLogRetrievalAsync capability) {
-                        return capability.start(new IPodLogRetrievalAsync.IPodLogListener() {
-                            @Override
-                            public void onOpen() {
-                            }
-
-                            @Override
-                            public void onMessage(String message) {
-                                listener.getLogger().print(message);
-                            }
-
-                            @Override
-                            public void onClose(int code, String reason) {
-                            }
-
-                            @Override
-                            public void onFailure(IOException e) {
-                                // If follow fails, try to restart it on the next loop.
-                                needToFollow.compareAndSet(false, true);
-                            }
-                        }, new IPodLogRetrievalAsync.Options()
-                                .follow()
-                                .container(container));
-                    }
-                }, null);
+              List<IPod> pods = client.list(ResourceKind.POD, getNamespace(overrides));
+              for (IPod pod : pods) {
```

Comment:
On whether the BC should produce a Pod, I did look into that ...the IBuildConfig surfaced by the restclient does not surface the necessary information in that regard.  To me at least, it was low enough priority to not deal with it at this time.  Among other things, I'd rather not pull in a SNAPSHOT version of the restclient prior to 3.4 shipping if I can help it.

Let me know how that affects your opinion.

I'll tinker with your timing window in the interim.